### DISCUSSION TITLE
DsfrDataTable : bug sur le « tout sélectionner »

### DISCUSSION THREAD

**Discussion by CopperGiraffe** [dnum-mi/vue-dsfr#939] [Ideas]
Salut l’équipe vue-dsfr,

J’ai relevé trois petits soucis sur la partie selectableRows :

1) Le « tout sélectionner » ne fonctionne pas comme attendu lorsque l’attribut `row-key` ne correspond a aucune `key` de la définition des colonnes. Cela peut arriver lorsqu’on a besoin d’accéder l’objet entier plutôt qu’à l’attribut au sein du slot `#cell`, on utilise dans ce cas une `key` dans la définition de notre colonne qui ne correspond pas à un champ de notre objet. 

2) Je constate que même lorsqu’il fonctionne comme attendu, la case « tout sélectionner » reste côché même lorsque l’on désélectionne une des lignes.

3) La sélection qu’on récupère via le v-model reste correcte lorsqu’on utilise de la pagination, mais visuellement les lignes restent cochés sur les autres pages du tableau alors qu’elle ne sont pas présentent dans la sélection.

Merci et bonne journée !

**Comment by laruiss** [comment]
En effet, nous devons retravailler ce composant.

**Reply by laruiss** [reply]
C’est corrigé
